src/kubernetes_charm.py

```diff
@@ -118,16 +128,17 @@ def _patch_service(self) -> None:
             spec=lightkube.models.core_v1.ServiceSpec(
                 ports=[
                     lightkube.models.core_v1.ServicePort(
-                        name="mysql-rw",
+                        name=self._SVC_NAME_READ_WRITE,
                         port=self._READ_WRITE_PORT,
-                        targetPort=self._READ_WRITE_PORT,
+                        targetPort=self._READ_WRITE_PORT,  # Dummy value if we use NodePort
                     ),
                     lightkube.models.core_v1.ServicePort(
-                        name="mysql-ro",
+                        name=self._SVC_NAME_READ_ONLY,
                         port=self._READ_ONLY_PORT,
-                        targetPort=self._READ_ONLY_PORT,
+                        targetPort=self._READ_ONLY_PORT,  # Dummy value if we use NodePort
                     ),
                 ],
+                type="NodePort" if self.is_exposed else "ClusterIP",
```

Comment:
Can you change service type with patch only (i.e. no need to replace)?

Reply:
It is a "patch" now. I tested it and the RW port was exposed using node port.